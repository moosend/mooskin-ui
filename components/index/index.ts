import '../_utils/globals/global.css';

export {default as Button} from '../Button';
export {default as CheckListItem} from '../CheckListItem';
export {default as CheckboxGroup, Checkbox} from '../Checkbox';
export {default as DatePicker} from '../DatePicker';
export {default as Form, FormGroup} from '../Form';
export {default as FileUpload} from '../File';
export {default as Fieldset} from '../Fieldset';
export {default as Grid, Row, Col} from '../Grid';
export {default as HorizontalRangeBar} from '../HorizontalRangeBar';
export {H1, H2, H3, H4, H5, H6} from '../Headings';
export {default as Input} from '../Input';
export {default as List, ListItem} from '../List';
export {default as LoadingBar} from '../LoadingBar';
export {default as Modal} from '../Modal';
export {default as Pagination} from '../Pagination';
export {default as RadioGroup, Radio} from '../Radio';
export {default as RadioAccordion, RadioAccordionContent} from '../RadioAccordion';
export {default as Steps, Step} from '../Steps';
export {default as SideBar, SidebarItem} from '../SideBar';
export {default as SmallIconButton} from '../SmallIconButton';
export {default as Select, Option} from '../Select';
export {default as Switch} from '../Switch';
export {default as StatsBox, StatsTitle, StatsNumber, StatsResult} from '../StatsBox';
export {default as TextArea} from '../TextArea';
export {default as TabbedContent, Content, Header, Tab} from '../TabbedContent';
export {default as Tags} from '../Tags';
export {default as TopNotification} from '../TopNotification';
export {default as Table, TableHeader} from '../Table';

export {Bar, Doughnut, Pie, Line} from '../Charts';
