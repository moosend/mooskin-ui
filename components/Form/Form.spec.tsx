import toJson from 'enzyme-to-json';
import * as React from 'react';
import {Button, Form, FormGroup, Input} from '../index';

import {mount, render, shallow} from 'enzyme';

describe('Form', () => {

    test('renders Form correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <Form className="myClass" dataLabel="SomeForm" onSubmit={func} style={{width: '50px'}}/>
        );

        expect(toJson(tree)).toMatchSnapshot();
    });

    test('renders FormGroup correctly', () => {

        const tree = shallow(
            <FormGroup className="myClass" style={{width: '50px'}} horizontal/>
        );

        expect(toJson(tree)).toMatchSnapshot();
    });

    test('renders Form with FormGroup correctly', () => {
        const func = jest.fn();

        const tree = mount(
            <Form className="myClass" dataLabel="SomeForm" onSubmit={func} style={{width: '50px'}}>
                <FormGroup className="myClass" style={{width: '50px'}} horizontal/>
            </Form>
        );

        expect(toJson(tree)).toMatchSnapshot();
    });

    test('renders properly with FormGroup and a child', () => {

        const component = mount(
            <Form className="myClass" dataLabel="plan">
                <FormGroup className="myClass" horizontal >
                    <Input />
                </FormGroup>
            </Form>
        );

        expect(component.find(Form).children.length).toBe(1);
        expect(component.find(FormGroup).children.length).toBe(1);
        expect(component.find(FormGroup).hasClass('myClass')).toBeTruthy;
    });

    test('renders properly with multiple children', () => {

        const component = shallow(
            <Form className="myClass" dataLabel="plan">
                <FormGroup className="myClass" horizontal >
                    <FormGroup>
                        <Input />
                        <Input />
                        <Input />
                    </FormGroup>
                    <FormGroup>
                        <Input />
                        <Input />
                        <Input />
                    </FormGroup>
                </FormGroup>
            </Form>
        );

        expect(component.find(Form).children.length).toBe(1);
        expect(component.find(FormGroup).length).toBe(3);
        expect(component.find(Input).length).toBe(6);
    });

    test('onSubmit callback function called when button is clicked', () => {
        const func = jest.fn();

        const component = shallow(
            <Form onSubmit={func} style={{color: 'blue'}}>
                <FormGroup>
                    <Input />
                    <Input />
                </FormGroup>
                <Button type="submit"/>
            </Form>
        );

        component.find(Button).simulate('click', { preventDefault: () => undefined });
        expect(func).toHaveBeenCalled();
    });

    // test('onSubmit callback ignored when button type is not submit, the custom button callback is called', () => {
    //     const func = jest.fn();
    //     const func2 = jest.fn();

    //     const component = shallow(
    //         <Form onSubmit={func} style={{color: 'blue'}}>
    //             <FormGroup>
    //                 <Input />
    //                 <Input />
    //             </FormGroup>
    //             <Button type="reset" onClick={func2}/>
    //         </Form>
    //     );

    //     component.find(Button).simulate('click');
    //     expect(func2).toHaveBeenCalled();
    // });

});
