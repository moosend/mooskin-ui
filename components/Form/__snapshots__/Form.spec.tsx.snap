// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders Form correctly 1`] = `
<div
  className="form-component container"
  style={
    Object {
      "width": "50px",
    }
  }
>
  <form
    className="form form myClass"
    name="nodejs"
  />
</div>
`;

exports[`Form renders Form with FormGroup correctly 1`] = `
<Form
  className="myClass"
  dataLabel="SomeForm"
  onSubmit={[Function]}
  style={
    Object {
      "width": "50px",
    }
  }
>
  <div
    className="form-component container"
    style={
      Object {
        "width": "50px",
      }
    }
  >
    <form
      className="form form myClass"
      name="nodejs"
    >
      <FormGroup
        className="myClass"
        horizontal={true}
        key="0"
        style={
          Object {
            "width": "50px",
          }
        }
      >
        <div
          className="formgroup-component container horizontal"
          style={
            Object {
              "width": "50px",
            }
          }
        />
      </FormGroup>
    </form>
  </div>
</Form>
`;

exports[`Form renders FormGroup correctly 1`] = `
<div
  className="formgroup-component container horizontal"
  style={
    Object {
      "width": "50px",
    }
  }
/>
`;
